interop-testing/src/main/java/io/grpc/testing/integration/Http2Client.java

```diff
@@ -0,0 +1,336 @@
+              + validTestCasesHelpText()
+      );
+      System.exit(1);
+    }
+  }
+
+  private void setUp() {
+    channel = createChannel();
+    blockingStub = TestServiceGrpc.newBlockingStub(channel);
+    asyncStub = TestServiceGrpc.newStub(channel);
+  }
+
+  private synchronized void tearDown() {
+    try {
+      if (channel != null) {
+        channel.shutdown();
+      }
+    } catch (RuntimeException ex) {
+      throw ex;
+    } catch (Exception ex) {
+      throw new RuntimeException(ex);
+    }
+  }
+
+  private void run() {
+    System.out.println("Running test " + testCase);
+    try {
+      runTest(Http2TestCases.fromString(testCase));
+    } catch (RuntimeException ex) {
+      throw ex;
+    } catch (Exception ex) {
+      throw new RuntimeException(ex);
+    }
+    System.out.println("Test completed.");
+  }
+
+  private void runTest(Http2TestCases testCase) throws Exception {
+    switch (testCase) {
+      case RST_AFTER_HEADER:
+        tester.rstStreamAfterHeader();
+        break;
+      case RST_AFTER_DATA:
+        tester.rstStreamAfterData();
+        break;
+      case RST_DURING_DATA:
+        tester.rstStreamDuringData();
+        break;
+      case GOAWAY:
+        tester.goAway();
+        break;
+      case PING:
+        tester.ping();
+        break;
+      case MAX_STREAMS:
+        tester.maxStreams();
+        break;
+      default:
+        throw new IllegalArgumentException("Unknown test case: " + testCase);
+    }
+  }
+
+  private class Tester {
+    private final SimpleRequest simpleRequest = SimpleRequest.newBuilder()
+        .setResponseSize(314159)
+        .setResponseType(PayloadType.COMPRESSABLE)
+        .setPayload(Payload.newBuilder()
+            .setBody(ByteString.copyFrom(new byte[271828])))
+        .build();
+
+    public void rstStreamAfterHeader() throws Exception {
+      try {
+        blockingStub.unaryCall(simpleRequest);
+        fail("Expected call to fail");
+      } catch (StatusRuntimeException ex) {
+        assertEquals(Status.Code.UNAVAILABLE, ex.getStatus().getCode());
+      }
+    }
+
+    public void rstStreamAfterData() throws Exception {
+      try {
+        blockingStub.unaryCall(simpleRequest);
+        fail("Expected call to fail");
+      } catch (StatusRuntimeException ex) {
+        assertEquals(Status.Code.UNAVAILABLE, ex.getStatus().getCode());
+      }
+
+      // Use async stub to verify data was received.
+      StreamRecorder<SimpleResponse> responseObserver = StreamRecorder.create();
+      asyncStub.unaryCall(simpleRequest, responseObserver);
+      responseObserver.awaitCompletion();
+      assertEquals(1, responseObserver.getValues().size());
+      assertEquals(Status.Code.UNAVAILABLE,
+          Status.fromThrowable(responseObserver.getError()).getCode());
+    }
+
+    public void rstStreamDuringData() throws Exception {
+      try {
+        blockingStub.unaryCall(simpleRequest);
+        fail("Expected call to fail");
```

Comment:
I'm very curious why JUnit methods are being used in a non JUnit environment.  Can these be avoided?